packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart

```diff
@@ -141,19 +144,56 @@ class MethodChannelShare extends SharePlatform {
     String? subject,
     String? text,
     Rect? sharePositionOrigin,
-  }) {
-    final mimeTypes =
-        files.map((e) => e.mimeType ?? _mimeTypeForPath(e.path)).toList();
+  }) async {
+    final filesWithPath = await _getFiles(files);
+
+    final mimeTypes = filesWithPath
+        .map((e) => e.mimeType ?? _mimeTypeForPath(e.path))
+        .toList();
 
     return shareFilesWithResult(
-      files.map((e) => e.path).toList(),
+      filesWithPath.map((e) => e.path).toList(),
       mimeTypes: mimeTypes,
       subject: subject,
       text: text,
       sharePositionOrigin: sharePositionOrigin,
     );
   }
 
+  /// if file doesn't contain path
+  /// then make new file in TemporaryDirectory and return with path
+  ///
+  /// the system will automatically delete files in this
+  /// TemporaryDirectory as disk space is needed elsewhere on the device
+  Future<List<XFile>> _getFiles(List<XFile> files) async {
+    int count = 1;
+
+    if (!files.every((element) => element.path.isNotEmpty)) {
```

Comment:
this if condition can be removed, `every` is already running through the whole array, so no need to do that twice

Reply:
it would be better to change every to any and negate the condition